### PR TITLE
fix(ci,test): setup-uvアクション化・PII防御・テスト強化 (PR #340 review)

### PR DESCRIPTION
**Pull request by yuta158**
## 概要
PR #340 のレビュー指摘対応。CI安全性・型精度・PII防御・テスト品質の4点を改善。

## 変更内容
- **ci**: setup-uvアクション(pinned SHA)への移行でsupply chain attack対策（3ジョブ）
- **security**: timeout/unexpected_errorログからstr(e)を除去→qualname化でPII漏洩防止
- **test**: asyncio.sleepをAsyncMockでモック化しリトライテストを高速・決定論化
- **type**: `__aexit__`型注釈を`Any→BaseException|None`に強化、TracebackType導入
- **test**: edge caseテスト追加（.github repo名、missing headerデフォルト、main()RuntimeError伝播、timeout/unexpected_errorの例外チェーン切断検証）
- **docs**: api_client.pyのdocstringをwhitespace-only検証実装と一致させる

## テスト
- [x] ローカルテスト完了（961 passed / 95.50% coverage）
- [x] ruff 0 errors（staged files）
- [x] mypy 0 errors
- [x] CI/CDパイプライン確認予定

## 関連PR
Refs #340 (PR)

---

### Commit 5ea09ac: NetworkError retry & PII防止

**変更対象**: `utils/github_client.py` + `tests/unit/test_github_client.py`

| 項目 | 旧仕様 | 新仕様 | 影響度 |
|-----|--------|--------|--------|
| **NetworkError処理** | 即失敗 (ConnectError/ReadError等) | TimeoutExceptionと同等のretry戦略を適用 | 🟡 Medium |
| **retry回数** | timeout: 5回まで | Network error: 5回まで | 🟡 Medium |
| **backoff戦略** | timeout専用 | timeout+network共通: exponential_backoff_with_jitter | 🟢 Low |
| **外部API側の見え方** | 同じ (GitHubAPIError) | 同じ (GitHubAPIError) | 🟢 Low |
| **PII漏洩防止** | except内raise（__context__に例外残存） | except外raise + from None（チェーン完全切断） | ✅ Security |
| **ログフィールド** | 既存: request_timeout | 新規: request_network_error（同一フォーマット） | 🟡 Medium |

**移行手順**: 
- 既存コード: ConnectError/ReadError/RemoteProtocolErrorはGitHubAPIError即失敗 → 即座にリトライを止める実装あれば影響あり。ただし、httpx.TransportError + retryは標準パターンのため、大半のコード（retry条件がない限り）は挙動変わらず。
- 新仕様: 自動retry → エラー率低減。ログには `request_network_error` イベントが追加される（既存監視ダッシュボードは変更不要、新rule追加で `request_network_error` を集計可能）。

**リスク要因**: なし（例外チェーン切断はセキュリティ向上、retry上限は既存と同じ）。

---

このPRは `/push-pr` スキルで自動生成されました。

### CODE REVIEW

**Comment by yuta158** [comment]
🔗 新規コミット追加: `33b8711` fix(client): PII漏洩防止・例外チェーン管理改善 (PR #341 review)

品質ゲート: 954 tests passed / 95.44% coverage / ruff 0 errors / mypy 0 errors

**Review by yuta158 — COMMENTED** [review]
## Implementation Review

### Verdict: FAIL

### Critical Findings
- utils/github_client.py:458 and tests/unit/test_github_client.py:1617 keep body_preview as raw response.content, and the new test explicitly asserts that a secret-like string is present in the logged preview. If an error response ever echoes a token, repo name, or other sensitive payload, it will be emitted into stdout/debug logs. Redact or omit body_preview before logging, and update the test to assert the redacted behavior. Blocking.

### Recommendations
- No additional blocking issues found in api_client.py, the settings tests, or the CI pinning change.

### Plan Drift / Scope
- No plan drift detected from the review body.

### Residual Risks
- The timeout and unexpected-error paths now rely on from None plus structured fields for diagnostics. Keep the targeted tests so future changes do not reintroduce leaked exception chains.

### Areas of Concern Response
- PII reduction is improved for exception chaining, but the response-body preview path still leaks raw data.

File: tests/unit/test_github_client.py [hunk 1]
```diff
@@ -1535,7 +1616,6 @@ async def test_httpx_status_error_429_missing_reset_header_falls_back_to_zero()
     assert route.call_count == 1
```
Comment (yaoki-dev) [inline tests/unit/test_github_client.py:1733]: tests/unit/test_github_client.py:1617 - body_preview に raw response.content をそのまま入れると、エラー応答がトークンや private repo 名を含む場合に stdout/debug logs へ漏れます。body_preview を redact するか、要約/ハッシュに置き換えてください。Blocking.

**Comment by yuta158** [comment]
@claude 

## 対応完了 (commit `b30db41`)

**マージブロッカー対応済み**: `max_retries` docstring に NetworkError を追記しました。

```python
# 修正後 (utils/github_client.py L156)
max_retries: 最大試行回数（5xx・timeout・NetworkError の再試行回数、初回含む）。
    デフォルト設定(timeout=30, max_retries=3)での最悪ケース: 約96秒。
```

SLA最悪ケース（約96秒）も同時に追記し、呼び出し元がタイムアウト計画を立てられるよう改善しました。

品質ゲート: 963 tests passed / 95.49% coverage / ruff 0 errors / mypy 0 errors

**Comment by yuta158** [comment]
@claude 

## 対応完了 (commit `b30db41`)

6件の品質改善推奨について対応しました。

| # | 指摘 | 対応 |
|---|------|------|
| 1 | `_request` Raises docstring NetworkError欠落 | ✅ 追記済み（前コミット） |
| 2 | `_handle_5xx_response` `from None` なし | ✅ 追記済み（前コミット） |
| 3 | SLA最悪ケース応答時間未記載 | ✅ `max_retries` docstringに「約96秒」追記 |
| 4 | `test_response_not_read_...` 内部実装依存 | ✅ `respx.mock` パターンに置換 |
| 5 | 404 `__context__ is None` 検証欠落 | ✅ アサーション追加 |
| 6 | `timeout_error_type`/`network_error_type` 統合 | ✅ 統合済み（前コミット） |

加えて、コードレビューフィードバックに基づき `_handle_retryable_error` を `_log_and_sleep_for_retry` にリネームし、SRP準拠（ログ・sleep専任）・keyword-only引数・`-> None` 返却に変更しました。

品質ゲート: 963 tests passed / 95.49% coverage / ruff 0 errors / mypy 0 errors
